velox/expression/tests/ExpressionFuzzer.cpp

```diff
@@ -1159,6 +1162,89 @@ void ExpressionFuzzer::go() {
   logStats();
 }
 
+void ExpressionFuzzer::errorExit(const std::string& text) {
+  VELOX_FAIL(text);
+}
+
+void ExpressionFuzzer::drilldown(
+    core::TypedExprPtr plan,
+    const RowVectorPtr& rowVector,
+    const std::vector<column_index_t>& columnsToWrapInLazy) {
+  if (try2(plan, rowVector, columnsToWrapInLazy)) {
+    errorExit("Retry should have failed");
+  }
+  bool minimalFound = false;
+  drilldownRecursive(plan, rowVector, columnsToWrapInLazy, minimalFound);
+  if (minimalFound) {
+    errorExit("Found minimal failing expression");
+  } else {
+    errorExit("Only the top level expression failed");
+  }
+}
+
+void ExpressionFuzzer::drilldownRecursive(
+    core::TypedExprPtr plan,
+    const RowVectorPtr& rowVector,
+    const std::vector<column_index_t>& columnsToWrapInLazy,
+    bool& minimalFound) {
+  bool anyFailed = false;
+  for (auto& input : plan->inputs()) {
+    if (!try2(input, rowVector, columnsToWrapInLazy)) {
+      anyFailed = true;
+      drilldownRecursive(input, rowVector, columnsToWrapInLazy, minimalFound);
+      if (minimalFound) {
+        return;
+      }
+    }
+  }
+  if (!anyFailed) {
+    minimalFound = true;
+    LOG(INFO) << "Failed with all children succeeding: " << plan->toString();
+    // Re-running the minimum failed. Put breakpoint here to debug.
+    try2(plan, rowVector, columnsToWrapInLazy);
+    if (!columnsToWrapInLazy.empty()) {
+      LOG(INFO) << "Trying without lazy:";
+      if (try2(plan, rowVector, {})) {
+        LOG(INFO) << "Minimal failure succeeded without lazy vectors";
+      }
+    }
+  }
+}
+
+bool ExpressionFuzzer::try2(
+    core::TypedExprPtr plan,
+    const RowVectorPtr& rowVector,
+    const std::vector<column_index_t>& columnsToWrapInLazy) {
+  VectorPtr result;
+  bool emptyResult = try1(plan, rowVector, columnsToWrapInLazy, result);
+  result = vectorFuzzer_.fuzzFlat(plan->type());
+  bool filledResult = try1(plan, rowVector, columnsToWrapInLazy, result);
+  if (emptyResult != filledResult) {
+    LOG(ERROR) << fmt::format(
```

Comment:
I think we can output a more helpful message , that passing in a flat result changed eval and needs investigation.